File: ios/Plugin/StoreReviewPlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

// Define the plugin using the CAP_PLUGIN Macro, and
// each method the plugin supports using the CAP_PLUGIN_METHOD macro.
CAP_PLUGIN(StoreReviewPlugin, "StoreReview",
           CAP_PLUGIN_METHOD(navigateToAppStore, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(requestReview, CAPPluginReturnPromise);
)

File: ios/Plugin/StoreReviewPlugin.swift
import Capacitor
import Foundation

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(StoreReviewPlugin)
public class StoreReviewPlugin: CAPPlugin {
    private let implementation = StoreReview()

    private var appId: String = ""

    @objc override public func load() {
        super.load()

        appId = getConfigValue("appId") as! String
    }

    @objc func navigateToAppStore(_ call: CAPPluginCall) {
        implementation.navigateToAppStore(appId) { res in
            call.resolve(["value": res])
        }
    }

    @objc func promtForRating(_ call: CAPPluginCall) {
        implementation.promtForRating {
            call.resolve()
        }
    }
}
